main.nf

```diff
@@ -118,6 +118,43 @@ process plot_quality {
     """
 }
 
+if(params.containsKey("cutadapt_params")) {
+    cutadapt_params_str = "${params.cutadapt_params.join(' ')}"
+    process cutadapt {
+
+        label 'med_cpu_mem'
+
+        input:
+            tuple sampleid, file(I1), file(I2), file(R1), file(R2) from to_fastq_filters
+        output:
+            tuple sampleid, file(I1), file(I2), file("${sampleid}_R1_trimmed.fq.gz"), file("${sampleid}_R2_trimmed.fq.gz") into to_barcodecop
+            file("${sampleid}.cutadapt.json") into cutadapt_json
+            file("${sampleid}.cutadapt.tsv") into cutadapt_log
+
+        publishDir "${params.output}/cutadapt/", overwrite: true, mode: 'copy', pattern: '*.{json,tsv}'
+
+        """
+        cutadapt ${cutadapt_params_str} -o ${sampleid}_R1_trimmed.fq.gz -p ${sampleid}_R2_trimmed.fq.gz ${R1} ${R2} --json=${sampleid}.cutadapt.json --report=minimal > ${sampleid}.cutadapt.tsv
+        """
+    }
+
+    process cutadapt_counts_concat {
+
+    input:
+        file("*.cutadapt.tsv") from cutadapt_log.collect()
+
+    output:
+        file("cutadapt_counts.csv") into cutadapt_counts_concat
+
+    """
+    stack_cutadapt_counts.sh *.cutadapt.tsv | sed -e 's/out_reads/cutadapt/' | xsv select -d '\t' sampleid,cutadapt  > cutadapt_counts.csv
```

Comment:
Is there a reason not to move all of the necessary processing into ``stack_cutadapt_counts.sh`` and write directly to an output file?

Reply:
Ok, not important to make any changes now. May revisit later.